src/bin/pg_autoctl/pgsql.c

```diff
@@ -1279,21 +1277,32 @@ parsePgsrSyncStateAndWAL(void *ctx, PGresult *result)
 	}
 }
 
+
 /*
- * pgsql_get_wal_lag_from_standby queries a standby PostgreSQL server to get the
- * replication lag as seen from the pg_stat_wal_receiver system view.
+ * pgsql_get_received_lsn_from_standby queries a standby PostgreSQL server to get the
+ * received_lsn value from the pg_stat_wal_receiver system view.
+ *
+ * received_lsn is the latest lsn known to be received and flushed to the disk. It does
+ * not specify if it is applied or not. Caller should have allocated necessary memory
+ * for result value.
+ *
+ * We are collecting the latest WAL entry that is received successfully. It will be
+ * eventually applied to the receiving database.  This information will later be
+ * used by monitor to decide which secondary has the latest data.
  */
 bool
-pgsql_get_wal_lag_from_standby(PGSQL *pgsql, int64_t *walLag)
+pgsql_get_received_lsn_from_standby(PGSQL *pgsql, char *receivedLSN, int bufferLength)
 {
 	SingleValueResultContext context;
 	char *sql =
-		"SELECT coalesce(latest_end_lsn - received_lsn, -1) "
+		"SELECT received_lsn"
 		"  FROM pg_stat_wal_receiver";
 
-	context.resultType = PGSQL_RESULT_BIGINT;
+	context.resultType = PGSQL_RESULT_STRING;
 	context.parsedOk = false;
 
+	log_trace("pgsql_get_received_lsn_from_standby : running %s", sql);
```

Comment:
I think the SQL statement is going to be logged by the `sql__execute_with_params` function at the DEBUG level anyway. Do we need this `log_trace` call here?

Reply:
removed